chalk-solve/src/clauses/builtin_traits/pointee.rs

```diff
@@ -0,0 +1,153 @@
+use crate::clauses::ClauseBuilder;
+use crate::rust_ir::WellKnownTrait;
+use crate::{Interner, RustIrDatabase, TraitRef};
+use chalk_ir::{
+    AliasTy, Floundered, Normalize, ProjectionTy, Substitution, Ty, TyKind, TyVariableKind,
+};
+
+use super::last_field_of_struct;
+
+fn push_clauses<I: Interner>(
+    db: &dyn RustIrDatabase<I>,
+    builder: &mut ClauseBuilder<'_, I>,
+    self_ty: Ty<I>,
+    metadata: Ty<I>,
+) {
+    let interner = db.interner();
+    let trait_id = db.well_known_trait_id(WellKnownTrait::Pointee).unwrap();
+    let substitution = Substitution::from1(interner, self_ty);
+    builder.push_fact(TraitRef {
```

Comment:
aren't we pushing this trait goal twice?

Reply:
![image](https://user-images.githubusercontent.com/45197576/234354552-c5f05e46-8ddc-42ca-86b0-762d5b7ef88e.png)
![image](https://user-images.githubusercontent.com/45197576/234354554-fc791c09-ec9d-437f-8410-dec39f7c9dde.png)